SYCL/ESIMD/api/functional/ctors/ctor_copy.cpp

```diff
@@ -77,35 +77,73 @@ class const_ref {
   static std::string get_description() { return "const reference"; }
 
   template <typename DataT, int NumElems>
-  static simd<DataT, NumElems> call_simd_ctor(const DataT *ref_data) {
+  static void call_simd_ctor(const DataT *ref_data, DataT *out) {
     simd<DataT, NumElems> source_simd;
     source_simd.copy_from(ref_data);
-    return call_simd_by_const_ref<DataT, NumElems>(
-        simd<DataT, NumElems>(source_simd));
+    call_simd_by_const_ref<DataT, NumElems>(simd<DataT, NumElems>(source_simd),
+                                            out);
   }
 
 private:
   template <typename DataT, int NumElems>
-  static simd<DataT, NumElems>
-  call_simd_by_const_ref(const simd<DataT, NumElems> &simd_by_const_ref) {
-    return simd_by_const_ref;
+  static void
+  call_simd_by_const_ref(const simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *out) {
+    simd_by_const_ref.copy_to(out);
   }
 };
 
-int main(int argc, char **argv) {
-  sycl::queue queue{esimd_test::ESIMDSelector{},
-                    esimd_test::createExceptionHandler()};
+// Struct that calls simd in provided context and then verifies obtained result.
+struct Verifier {
+  template <typename DataT, int NumElems, typename TestCaseT>
+  static bool verify(sycl::queue &queue, const std::vector<DataT> ref_data,
+                     const std::string &data_type) {
+    bool passed = true;
+
+    shared_vector<DataT> result(NumElems, shared_allocator<DataT>{queue});
+    shared_vector<DataT> shared_ref_data(ref_data.begin(), ref_data.end(),
+                                         shared_allocator<DataT>{queue});
+
+    queue.submit([&](sycl::handler &cgh) {
+      const auto ref = shared_ref_data.data();
+      auto out = result.data();
```

Comment:
```suggestion
      const DataT * const ref = shared_ref_data.data();
      const auto out = result.data();
```
The `const auto` gives the `DataT * const` type actually

Reply:
I've applied this comment in this commit: https://github.com/intel/llvm-test-suite/pull/614/commits/e87ad2ec4b8cff0070719ae8992d50d244c2a019

Reply:
Looks like the `out` is still `auto` instead of the `const auto`, and I don't see any reasoning here. Is it intended or did it just slip by? Applies to `ctor_def` as well.